grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx

```diff
@@ -1,54 +1,241 @@
 
   return (
-    <form onSubmit={handleSubmit(onSubmit)}>
-      <Controller
-        name={'onCallApiUrl'}
-        control={control}
-        rules={{ required: 'OnCall API URL is required', validate: validateURL }}
-        render={({ field }) => (
-          <Field
-            key={'Name'}
-            label={'OnCall API URL'}
-            invalid={Boolean(formState.errors.onCallApiUrl)}
-            error={formState.errors.onCallApiUrl?.message}
-          >
-            <Input {...field} placeholder={'OnCall API URL'} data-testid="oncall-api-url-input" />
-          </Field>
-        )}
-      />
-      <HorizontalGroup>
-        <Button type="submit" disabled={!formState.isValid}>
-          Test & Save connection
-        </Button>
-        {config.featureToggles.externalServiceAccounts && <Button variant="secondary">Recreate service account</Button>}
-      </HorizontalGroup>
-    </form>
+    <>
+      <Text.Title level={3} className="u-margin-bottom-md">
+        Configure Grafana OnCall
+      </Text.Title>
+      {props.plugin.meta.enabled && <PluginConfigAlert />}
+      {getIsRunningOpenSourceVersion() ? <OSSPluginConfigPage {...props} /> : <CloudPluginConfigPage />}
+    </>
   );
 });
+
+const CloudPluginConfigPage = observer(() => {
+  const {
+    pluginStore: { isPluginConnected },
+  } = rootStore;
+  const styles = useStyles2(getStyles);
+
+  return (
+    <>
+      <Text type="secondary" className={styles.secondaryTitle}>
+        This is a cloud-managed configuration.
+      </Text>
+      {!isPluginConnected && <Button onClick={() => window.open(REQUEST_HELP_URL, '_blank')}>Request help</Button>}
+    </>
+  );
+});
+
+const OSSPluginConfigPage = observer(
+  ({ plugin: { meta } }: PluginConfigPageProps<PluginMeta<OnCallPluginMetaJSONData>>) => {
+    const {
+      pluginStore: {
+        updatePluginSettingsAndReinitializePlugin,
+        connectionStatus,
+        recreateServiceAccountAndRecheckPluginStatus,
+        isPluginConnected,
+      },
+      loaderStore,
+    } = rootStore;
+    const styles = useStyles2(getStyles);
+    const { handleSubmit, control, formState } = useForm<PluginConfigFormValues>({
+      mode: 'onChange',
+      defaultValues: { onCallApiUrl: getOnCallApiUrl(meta) },
+    });
+    const isReinitializating = loaderStore.isLoading(ActionKey.UPDATE_SETTINGS_AND_REINITIALIZE_PLUGIN);
+    const isRecreatingServiceAccount = loaderStore.isLoading(ActionKey.RECREATE_SERVICE_ACCOUNT);
+
+    const onSubmit = async (values: PluginConfigFormValues) => {
+      await updatePluginSettingsAndReinitializePlugin({ ...meta.jsonData, onCallApiUrl: values.onCallApiUrl });
+    };
+
+    return (
+      <>
+        <Text type="secondary" className={styles.secondaryTitle}>
+          This page will help you to connect OnCall backend and OnCall Grafana plugin.
+        </Text>
+        <CollapsibleTreeView
+          className={styles.treeView}
+          configElements={[
+            {
```

Comment:
maybe worth splitting these items somehow? seems like a lot of nesting/conditionals